axelor-base/src/main/java/com/axelor/apps/base/db/repo/PartnerBaseRepository.java

```diff
@@ -69,6 +69,14 @@ public Partner save(Partner partner) {
               I18n.get(IExceptionMessage.PARTNER_EMAIL_EXIST));
         }
       }
+
+      for (PartnerAddress partnerAddress : partner.getPartnerAddressList()) {
```

Comment:
If not contactPartner first set null to mainContact and then set mainAddress in loop